src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,166 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            linkLast(value);
+            return;
+        }
+        checkIndex(index);
+        Node<T> elBefore = searchNode(index).prev;
+        Node<T> elAfter = searchNode(index);
+        Node<T> newNode = new Node<>(elBefore, value, elAfter);
+        if (elBefore == null) {
+            head = newNode;
+        } else {
+            elBefore.next = newNode;
+            elAfter.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        Object[] arr = list.toArray();
+        Node<T> elBefore = tail;
+        for (Object obj: arr) {
+            T t = (T) obj;
+            Node<T> node = new Node<>(elBefore, t, null);
+            if (elBefore.prev == null) {
+                head = node;
+            } else {
+                elBefore.next = node;
+            }
+            elBefore = node;
+            size++;
+        }
+        tail = elBefore;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return searchNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        T oldValue = searchNode(index).value;
+        searchNode(index).value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        return unLink(searchNode(index));
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> node = head;
+        for (int i = 0; i < size; i++) {
+            if (object == node.value || object != null && object.equals(node.value)) {
+                unLink(node);
+                return true;
+            }
+            node = node.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkIndex(int index) {
+        if (index >= 0 && index < size) {
+            return;
+        } else {
+            throw new IndexOutOfBoundsException("Wrong index");
+        }
```

Comment:
you can simplify your method, add "not" logical operation, and replace return with your exception